### DISCUSSION TITLE
Is there a way to make a poster paper with Quarto?

### DISCUSSION THREAD

**Discussion by cooknl** [quarto-dev/quarto-cli#2205] [Q&A]
I'm trying to get away from slideshows, and I'm thinking about large-format poster papers as an alternative, in which the poster paper is "traversed" during the presentation, including zoom out and zoom in

**Comment by cooknl** [comment]
<https://github.com/brentthorne/posterdown> for RMarkdown is an example of what I mean for the source material, but then combine it with pan and zoom of the view.

**Comment by mcanouil** [comment]
For now, I am not aware of `posterdown` being ported to Quarto as formats extensions (see <https://quarto.org/docs/extensions/formats.html>).
I was thinking of doing it in the future if not done by someone else by then.
You could give it a try and/try define a new format with all the specs you want.

PS: would you mind changing the Discussion category from "ideas" to "Q&A"? Unless you are actually really asking "can someone do it?" rather than "how can I do it?".

**Reply by cderv** [reply]
When we'll work on paged.js integration (related to pagedown in R Markdown ecosystem), doing poster will be possible more easily by creating a format to work with Paged.js. **pagedown** has poster example for example. **posterdown** is using **pagedown** also. 
So I would wait for that.

I wonder if something cannot already be done with PDF as doing poster in LaTeX is possible, and Quarto extension already support creating special format.

**Reply by mcanouil** [reply]
I never used `posterdown` and did not realise it uses `pagedown` (I thought it was more of a LaTeX based R package 🤷‍♂️).

Yes, I used some LaTeX templates during my Ph.D., so the LaTeX format approach is definitely possible right now.

**Reply by cooknl** [reply]
> For now, I am not aware of `posterdown` being ported to Quarto as formats extensions (see https://quarto.org/docs/extensions/formats.html). I was thinking of doing it in the future if not done by someone else by then. You could give it a try and/try define a new format with all the specs you want.
> 
> PS: would you mind changing the Discussion category from "ideas" to "Q&A"? Unless you are actually really asking "can someone do it?" rather than "how can I do it?".

Changed category to Q&A...it was a double question

1. Is it possible
2. How to do it

I'll play around with a template

I can see creating a large format PDF, but then using a separate application for the interaction with the content (pan/zoom), but it would be cool to program transitions to allow focus on the presentation instead of controlling the movement in realtime

**Reply by cooknl** [reply]
Also, separately, only with desktop site (not GitHub mobile app, nor mobile site) was is clear how to change the label from Idea to Q&A

**Reply by mcanouil** [reply]
The animation part can be achieved with JavaScript/CSS, thus HTML.
You could get a page layout without Paged.js, but it would be a bit like reinventing the wheel.

**Reply by mcanouil** [reply]
> Also, separately, only with desktop site (not GitHub mobile app, nor mobile site) was is clear how to change the label from Idea to Q&A

App/Desktop version do not have all the features. This is true for any website-based services.

**Reply by jcolomb** [reply]
it may be just needed some tweaks for the beamer output using beamer-poster templates, see https://www.overleaf.com/latex/templates?q=beamer+poster for examples.

Probably getting a quarto template that would translate is some latex templates would be nice. Aso having one for the "betterland" template of posterdown would be cool (the existing one is a html template : https://github.com/brentthorne/posterdown/blob/master/inst/rmarkdown/templates/posterdown_betterland/resources/template.html).

I personally do not really see a reason to create html posters (websites and reveal.js presentations are better options for me).

**Reply by mcanouil** [reply]
For short:

- "how?" has been answered, i.e., "make a Quarto format extension".
- "how difficult?", the heavy lifting is already done in `posterdown`.
- "who?", well whoever wants to do it.
- "when?" when Paged.js is suported natively in Quarto (which is scheduled). Why? Because `posterdown` HTML poster format imports `pagedown`, thus Paged.js, and because making an extension for PDF and HTML from the start is more efficient than doing some kind of patchwork or refactoring the whole extension. Also, HTML + CSS is by far more flexible and powerful than LaTeX, because in a world of hybrid and online conferences, interactivity is key, but this is not possible in static LaTeX PDF.

I might be the one to do it, but definitely not before Paged.js native support.

**Reply by awav** [reply]
@mcanouil have you managed to work on that? Any news about Paged.js support?

**Reply by mcanouil** [reply]
Not at all, because there isn't native support for Paged.js right now.

**Comment by cooknl** [comment]
My initial intent was to create a large form factor area for static or
interactive content, that could then be navigated during a "prezi-like"
presentation, or that could be explored by someone on their own time,
without the limitations of the 16:9 frame and the library constraints of a
slide deck

NLC

Mickaël Canouil ***@***.***> schrieb am Do., 8. Sept. 2022,
10:29:

**Reply by mcanouil** [reply]
I got that and I think this is definitely something worth adding to Quarto via extension.
To note, Paged.js can make "page" of whatever dimensions.

**Comment by TS-CUBED** [comment]
Are there any news on this? Since RMarkdown is reasonably close to Quarto, I could get away with moving to Posterdown for the poster only and then back to Quarto, but would be really nice to stay within quarto.

**Reply by mcanouil** [reply]
No updates means no updates.
It's plan but for the time being building your own Quarto extension format (especially when using LaTeX) is your best option.

**Comment by baptiste** [comment]
You might like to play with Typst; it's still early in development but I think for the purpose of placing content at arbitrary positions on a page it'll be much more convenient than LaTeX-based solutions (and I say this as someone who's [wandered relatively deep](https://b.bapt.xyz/posts/poster/) into that rabbit hole!). The grid layout system is still on their Todo list, however, so it'll probably remain a bit rough to make posters until then.

I don't know Typst yet, but I briefly tried it to reproduce this [minimal example](https://github.com/baptiste/quarto-flowfram#readme) and it was [pretty straightforward](https://typst.app/project/rRmHqfox3VtLzawFfLOcWm). Adapting it to quarto with a custom Typst template should be doable.

**Reply by TS-CUBED** [reply]
Wow, those Quarto posters in your blog look awesome. I don't use R, so would have to work out how to do that with Julia code.

**Reply by cderv** [reply]
@baptiste unfortunately the typst link is not public... would love to see your try !

**Reply by awav** [reply]
@cderv, I've recently started using typst, and it is a really flexible tool (although young). Here is a package for posters: https://github.com/pncnmnp/typst-poster. As you can see the final file with the code for the poster is quite concise.

**Reply by cderv** [reply]
Yes typst seems really good ! I plan to look further into it - glad you already did and was able to share an example! 
Regarding the private link I was refering to https://typst.app/project/p5nqRYBCD5gqZWZox5QtUC shared above.

**Reply by baptiste** [reply]
I fixed the link – but this was really just a basic experiment, nothing interesting. It's promising to port the full poster example though, but I haven't got to it yet.

**Comment by baptiste** [comment]
The other thing that comes to my mind from the OP (not so much a poster but more an infinite canvas) are tools such as Excalidraw, draw.io, and tl;draw. Each of these uses an open format (json/xml) that can be generated programmatically, and so a quarto document could potentially be translated into one of these outputs. 

I've made a few attempts for specific tasks, for example [minixcali](https://github.com/baptiste/minixcali#readme) and this application to [display all math equations on a canvas](https://b.bapt.xyz/posts/maths-excalidraw/), there's also [coolbutuseless' minidrawio](https://github.com/coolbutuseless/minidrawio) device, etc.

Specifying a layout for the elements is an interesting question, ideally it would be done in a feedback loop where one alters the position manually on the canvas and imports the positions back into the source document (again, since these are xml/json based formats, this is pretty straight-forward). I wonder what would be a good convention to store such positioning info.

As a side-note, pandoc has some support for ICML (Indesign), but I don't know if that helps much (requires Indesign, and again I don't think the positioning w/could be automated).

**Comment by higgi13425** [comment]
For many scientists, the general need is for a large (usually either 3 feet tall by 6 feet wide (US) or 1189 mm (46.8 inches) Height and 841 mm (33.1 inches) Width (Europe went with tall format). We generally print these on heavy glossy paper and put into poster tubes for travel, or some folks print them on cloth (so that they fold, easier to travel). It would be great to have a printable quarto version for this purpose, that could generally print both wide and tall formats, and have a few standard templates to start with.

**Reply by mcanouil** [reply]
See <https://github.com/quarto-ext/typst-templates/tree/main/poster>.

**Reply by danieltomasz** [reply]
The extenision is really nice, but how to to specify vertical A0 format? When I try  to set it vertical, it just cut half of the content; Probably I need to look into the  typst orginal extension adjust manually, but it would be great to know how to get it out of the box

**Reply by mcanouil** [reply]
> Probably I need to look into the typst orginal extension adjust manually, but it would be great to know how to get it out of the box

Indeed.

**Reply by higgi13425** [reply]
There are several examples, including euro A0 here - feel free to share - https://github.com/higgi13425/quarto_poster

**Reply by higgi13425** [reply]
@cooknl on the pan and zoom front - here is an experiment with scrollytelling - https://phiggins.quarto.pub/spending-wisely-on-ibd-medications/ which may help

**Reply by cooknl** [reply]
> @cooknl on the pan and zoom front - here is an experiment with scrollytelling - https://phiggins.quarto.pub/spending-wisely-on-ibd-medications/ which may help 

That has some potential

My primary idea is a single massive canvas with all content, albeit at different scales, that can be navigated either freely or in a predetermined sequence of pan and zooms

**Reply by higgi13425** [reply]
It would seem that a clever combination of the quarto typst poster above  https://github.com/higgi13425/quarto_poster and the scrollytelling capabilities, which rely on the `closeread` quarto extension here https://closeread.dev/ (code repo for the example here - https://github.com/higgi13425/scrollytelling_closeread) might get you there. Let me know if you figure out how to combine the two.

**Comment by engineer-pat** [comment]
I converted higgi13425's poster template to use Python: https://github.com/engineer-pat/quarto-poster-template

Credit to https://github.com/higgi13425/quarto_poster and Opus 4.8.
